### DISCUSSION TITLE
Modifying viewport meta-tag without ssrTemplate

### DISCUSSION THREAD

**Discussion by fflaten** [facebook/docusaurus#7944] [Q&A] [answered]
Is there any way besides `ssrTemplate` to **modify** the viewport meta-tag in `<head>` for all pages?

I would like to stretch header and footer to the edges ("safe-area") in landscape mobile view for iPhones, but AFAIK we need to append `viewport-fit=cover` in that tag to make it work. `ssrTemplates` makes development hard since `docusaurus start` doesn't use SSR.

**Comment by slorber** [comment]
There is another template in development, it's not configurable but we could make it configurable.

https://github.com/jantimon/html-webpack-plugin/blob/main/docs/template-option.md

I don't like it but currently we use 2 different templating languages for SSR and dev: eta vs ejs => we should probably only use just one before exposing this

---

What I would also try is to create a layout wrapper with this:

```js
export default function LayoutWrapper(props: Props): JSX.Element {
  return (
    <>
      <Head>
        <meta name="viewport" content="initial-scale=1, viewport-fit=cover" />
      </Head>
      <Layout {...props} />
    </>
  );
}
```

It doesn't replace the template meta data (so there will be twice the same meta) but hopefully the last one wins? 🤷‍♂️

**Reply by fflaten** [reply]
Thanks for the quick response. It might use the last one when duplicates, but that's probably browser-specific and something I don't want to rely on.

Ideally I'd be able to override it simply by using [themeConfig.metadata](https://docusaurus.io/docs/api/themes/configuration#metadata) like other meta-tags, but it also adds it as a duplicate just like a component or plugin will.

Having to maintain one or two templates in upgrades just to override a tag used for visuals/themes seems overkill.

**Reply by slorber** [reply]
>  It might use the last one when duplicates, but that's probably browser-specific and something I don't want to rely on.

technically, you can remove the duplicate one in prod by editing the ssr template, so it's only a dev issue

Please double-check first if it's not even in the spec 🤷‍♂️ it could be 

> Having to maintain one or two templates in upgrades just to override a tag used for visuals/themes seems overkill.

Agree on that, just trying to provide a good enough workaround.

---

I'm not so familiar with this meta and all its possible use-cases, can you explain yours better? 

Preferably illustrated with screenshots to make it easy to understand?

**Reply by slorber** [reply]
Looks like my analysis was wrong and react-helmet/react-helmet-async (Docusaurus `<Head>`) is able to deduplicate metadata.

Moving these meta out of the template should allow you to override them more easily: https://github.com/facebook/docusaurus/pull/7952

**Reply by fflaten** [reply]
> > Having to maintain one or two templates in upgrades just to override a tag used for visuals/themes seems overkill.
> 
> Agree on that, just trying to provide a good enough workaround.

I appreciate it 🙂 

> I'm not so familiar with this meta and all its possible use-cases, can you explain yours better?
> 
> Preferably illustrated with screenshots to make it easy to understand?

See https://webkit.org/blog/7929/designing-websites-for-iphone-x/ . Trying to solve [this](https://github.com/pester/docs/issues/223) specifically, but viewport can be used to limit zoom-range also.

**Comment by fflaten** [comment, marked as answer]
Easy solution implemented in #7952 which was released in v2.1. See https://docusaurus.io/blog/2022/09/01/docusaurus-2.1#overriding-default-meta-tags

Thanks again @slorber 👏
